Safari should pass current URL to ARQL when using ASVLaunchPreview

grorg opened the pull request:
#### 60f0cac6d918171da9beae365f712a3ac1166bd1
<pre>
Safari should pass current URL to ARQL when using ASVLaunchPreview
<a href="https://bugs.webkit.org/show_bug.cgi?id=250056">https://bugs.webkit.org/show_bug.cgi?id=250056</a>
rdar://103499310

Reviewed by Tim Horton.

ARQL have updated the ASVLaunchPreview API to allow
an originating URL.

* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/SystemPreviewController.h:

Canonical link: <a href="https://commits.webkit.org/258439@main">https://commits.webkit.org/258439@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/236e6008120b3b2dd7cca80519df4246fcc21ff4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101988 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/11133 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35059 "Failed to checkout and rebase branch from PR 8177") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/111320 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12103 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/2049 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94397 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/109075 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107769 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/43/builds/35059 "Failed to checkout and rebase branch from PR 8177") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/94397 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91148 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/43/builds/35059 "Failed to checkout and rebase branch from PR 8177") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94397 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/4715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/43/builds/35059 "Failed to checkout and rebase branch from PR 8177") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4806 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/1883 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/10881 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/43/builds/35059 "Failed to checkout and rebase branch from PR 8177") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/5791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/6555 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3045 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
```diff
@@ -281,15 +280,18 @@ - (UIImage *)previewController:(QLPreviewController *)controller transitionImage
 #if HAVE(UIKIT_WEBKIT_INTERNALS)
     url.removeFragmentIdentifier();
     NSURL *nsurl = (NSURL *)url;
-    if ([getASVLaunchPreviewClass() respondsToSelector:@selector(beginPreviewApplicationWithURLs:is3DContent:completion:)])
```
Do you want to keep the old path around?

grorg replied:
I'll check if it has been removed.

grorg replied:
It's still there so I'll add an `else if`

grorg committed:
Safari should pass current URL to ARQL when using ASVLaunchPreview
https://bugs.webkit.org/show_bug.cgi?id=250056
rdar://103499310

Reviewed by Tim Horton.

ARQL have updated the ASVLaunchPreview API to allow
an originating URL.

* Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm:
* Source/WebKit/UIProcess/SystemPreviewController.h:

Canonical link: https://commits.webkit.org/258439@main
